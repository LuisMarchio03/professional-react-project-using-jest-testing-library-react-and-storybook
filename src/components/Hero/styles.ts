import { styled } from '../../styles'

export const Container = styled('main', {
  backgroundSize: 'cover !important',
  backgroundPosition: 'center !important',
  backgroundBlendMode: 'overlay !important',
  backgroundRepeat: 'no-repeat !important',

  width: '100%',

  color: '$white',
  padding: '6.25rem 0',

  div: {
    width: '100%',
    padding: '0  8px',

    '@bp_md': {
      padding: '0  16px',
    },

    '@bp_xl': {
      width: '1140px',
      margin: '0 auto',
    },

    ul: {
      paddingLeft: '0',
      marginTop: '1.25rem',

      li: {
        '&::before': {
          content: '\\2713\\0020',
          color: '$yellow500',
        },
      },
    },

    button: {
      marginTop: '1.2rem',
    },
  },
})
