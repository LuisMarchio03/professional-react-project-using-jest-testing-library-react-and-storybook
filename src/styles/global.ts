import { globalCss } from '.'

export const globalStyles = globalCss({
  '*': {
    margin: 0,
    padding: 0,
    boxSizing: 'border-box',
  },

  body: {
    backgroundColor: '$white',
    color: '$gray600',
    '-webkit-font-smoothing': 'antialiased',
  },

  'body, input, textarea, button': {
    fontFamily: '$poppins, sans-serif',
    fontWeight: 300,
  },

  ul: {
    listStyle: 'none',
  },

  'h1, h2, h3, h4, h5, h6': {
    fontWeight: 700,
    letterSpacing: '0.05rem',
    lineHeight: '1.3',
  },

  h1: {
    fontSize: '$4xl',

    '@bp_lg': {
      fontSize: '$6xl',
    },
  },

  h2: {
    fontSize: '@3xl',

    '@bp_lg': {
      fontSize: '$5xl',
    },
  },

  h3: {
    fontSize: '$2xl',

    '@bp_lg': {
      fontSize: '$4xl',
    },
  },

  h4: {
    fontSize: '$xl',
    fontWeight: 600,

    '@bp_lg': {
      fontSize: '$3xl',
    },
  },

  h5: {
    fontSize: '$lg',
    fontWeight: 300,

    '@bp_lg': {
      fontSize: '$2xl',
    },
  },

  h6: {
    fontSize: '$md',
    fontWeight: 600,

    '@bp_lg': {
      fontSize: '$xl',
    },
  },
})
