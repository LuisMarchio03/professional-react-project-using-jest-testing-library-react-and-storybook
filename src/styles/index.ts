import { createStitches } from '@stitches/react'

export const {
  config,
  styled,
  css,
  globalCss,
  keyframes,
  getCssText,
  theme,
  createTheme,
} = createStitches({
  theme: {
    colors: {
      white: '#FFF',

      gray600: '#212121',
    },

    fontSizes: {
      md: '1.125rem',
      lg: '1.25rem',
      xl: '1.5rem',
      '2xl': '2rem',
    },

    fonts: {
      roboto: 'Roboto',
      poppins: 'Poppins',
    },
  },
})
