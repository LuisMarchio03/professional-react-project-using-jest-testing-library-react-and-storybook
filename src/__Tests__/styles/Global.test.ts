import { globalStyles } from '../../styles/global'
import { theme } from '../../styles'

describe('Global Styles', () => {
  test('should render global styles', () => {
    expect(globalStyles).toMatchSnapshot()
  })

  test('should the to match snapshot', () => {
    expect(theme).toMatchSnapshot()
  })
})
