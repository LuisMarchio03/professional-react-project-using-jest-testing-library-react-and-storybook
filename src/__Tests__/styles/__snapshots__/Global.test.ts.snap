// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Styles should render global styles 1`] = `[Function]`;

exports[`Global Styles should the to match snapshot 1`] = `
Object {
  "className": "t-mBMh",
  "colors": Object {
    "gray600": F {
      "prefix": "",
      "scale": "colors",
      "token": "gray600",
      "value": "#212121",
    },
    "red500": F {
      "prefix": "",
      "scale": "colors",
      "token": "red500",
      "value": "#ff3d00",
    },
    "red700": F {
      "prefix": "",
      "scale": "colors",
      "token": "red700",
      "value": "#b22a00",
    },
    "white": F {
      "prefix": "",
      "scale": "colors",
      "token": "white",
      "value": "#FFF",
    },
    "yellow300": F {
      "prefix": "",
      "scale": "colors",
      "token": "yellow300",
      "value": "#fff350",
    },
    "yellow500": F {
      "prefix": "",
      "scale": "colors",
      "token": "yellow500",
      "value": "#ffc107",
    },
    "yellow700": F {
      "prefix": "",
      "scale": "colors",
      "token": "yellow700",
      "value": "#c79100",
    },
  },
  "fontSizes": Object {
    "2xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "2xl",
      "value": "1.5rem",
    },
    "3xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "3xl",
      "value": "2rem",
    },
    "4xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "4xl",
      "value": "2.25rem",
    },
    "5xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "5xl",
      "value": "3.125rem",
    },
    "6xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "6xl",
      "value": "3.75rem",
    },
    "7xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "7xl",
      "value": "4.5rem",
    },
    "8xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "8xl",
      "value": "6rem",
    },
    "9xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "9xl",
      "value": "8rem",
    },
    "lg": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "lg",
      "value": "1.125rem",
    },
    "md": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "md",
      "value": "1rem",
    },
    "sm": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "sm",
      "value": "0.875rem",
    },
    "xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "xl",
      "value": "1.25rem",
    },
    "xs": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "xs",
      "value": "0.75rem",
    },
  },
  "fonts": Object {
    "poppins": F {
      "prefix": "",
      "scale": "fonts",
      "token": "poppins",
      "value": "Poppins",
    },
    "roboto": F {
      "prefix": "",
      "scale": "fonts",
      "token": "roboto",
      "value": "Roboto",
    },
  },
  "selector": ".t-mBMh",
  "toString": [Function],
}
`;

exports[`Global Styles should them to match snapshot 1`] = `
Object {
  "className": "t-mBMh",
  "colors": Object {
    "gray600": F {
      "prefix": "",
      "scale": "colors",
      "token": "gray600",
      "value": "#212121",
    },
    "red500": F {
      "prefix": "",
      "scale": "colors",
      "token": "red500",
      "value": "#ff3d00",
    },
    "red700": F {
      "prefix": "",
      "scale": "colors",
      "token": "red700",
      "value": "#b22a00",
    },
    "white": F {
      "prefix": "",
      "scale": "colors",
      "token": "white",
      "value": "#FFF",
    },
    "yellow300": F {
      "prefix": "",
      "scale": "colors",
      "token": "yellow300",
      "value": "#fff350",
    },
    "yellow500": F {
      "prefix": "",
      "scale": "colors",
      "token": "yellow500",
      "value": "#ffc107",
    },
    "yellow700": F {
      "prefix": "",
      "scale": "colors",
      "token": "yellow700",
      "value": "#c79100",
    },
  },
  "fontSizes": Object {
    "2xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "2xl",
      "value": "1.5rem",
    },
    "3xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "3xl",
      "value": "2rem",
    },
    "4xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "4xl",
      "value": "2.25rem",
    },
    "5xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "5xl",
      "value": "3.125rem",
    },
    "6xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "6xl",
      "value": "3.75rem",
    },
    "7xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "7xl",
      "value": "4.5rem",
    },
    "8xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "8xl",
      "value": "6rem",
    },
    "9xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "9xl",
      "value": "8rem",
    },
    "lg": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "lg",
      "value": "1.125rem",
    },
    "md": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "md",
      "value": "1rem",
    },
    "sm": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "sm",
      "value": "0.875rem",
    },
    "xl": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "xl",
      "value": "1.25rem",
    },
    "xs": F {
      "prefix": "",
      "scale": "fontSizes",
      "token": "xs",
      "value": "0.75rem",
    },
  },
  "fonts": Object {
    "poppins": F {
      "prefix": "",
      "scale": "fonts",
      "token": "poppins",
      "value": "Poppins",
    },
    "roboto": F {
      "prefix": "",
      "scale": "fonts",
      "token": "roboto",
      "value": "Roboto",
    },
  },
  "selector": ".t-mBMh",
  "toString": [Function],
}
`;
